### DISCUSSION TITLE
程式包(package)matplotlib裡的程式包-pylab

### DISCUSSION THREAD

**Discussion by enoch-lightning** [1101-GS4719-C/Python-Community-GS4719-C#284] [General]
上課老師說，兩個以上的*.py檔案(模組)，放在一個資料夾(目錄)，這個資料夾就稱為程式包。若一個程式包很大，代表裡面有很多功能。這個大程式包裡面就可能會有一些小的程式包。如matplotlib為python的第三方函式庫，其為一程式包。這程式包裡面用來畫圖的小程式包，最有名的莫非pyplot莫屬。除pyplot之外，有一小程式包名為pylab，可用於互動式繪圖。我目前不是很了解互動式繪圖是什麼意思，不知是否是如jupyter lab的操作，有很多個cell，我可以分別執行許多cell的意思。

**Comment by syt-110** [comment]
記得微積分python紫色的那本例題是用pylab，感覺可以去翻翻~(?

**Comment by enoch-lightning** [comment]
我目前的理解如下
pylab的功能就是把matploblib.pyplot的功能與numpy的數學計算功能整合在一起。pylab是附於matplotlib這個第三方函式庫裡面的模組(module)。言下之意就是，使用matplotlib.pyplot加上numpy就可以達到pylab的效果。而且，使用matploblib.pyplot加上numpy取代pylab也是比較建議的作法。原因我不知道是為什麼。

**Reply by enoch-lightning** [reply]
這邊附上參考的stackoverflow網頁。老師好像說有問題都可以上stackoverflow找答案。
[pylab究竟是什麼？](https://stackoverflow.com/questions/11469336/what-is-the-difference-between-pylab-and-pyplot)
